pyfolio/perf_attrib.py

```diff
@@ -120,133 +120,102 @@ def perf_attrib(returns,
             2017-01-01  0.249087  0.935925        1.185012          1.185012
             2017-01-02 -0.003194 -0.400786       -0.403980         -0.403980
     """
-    missing_stocks = positions.columns.difference(
-        factor_loadings.index.get_level_values(1).unique()
-    )
-
-    # cash will not be in factor_loadings
-    num_stocks = len(positions.columns) - 1
-    missing_stocks = missing_stocks.drop('cash')
-    num_stocks_covered = num_stocks - len(missing_stocks)
-    missing_ratio = round(len(missing_stocks) / num_stocks, ndigits=3)
-
-    if num_stocks_covered == 0:
-        raise ValueError("Could not perform performance attribution. "
-                         "No factor loadings were available for this "
-                         "algorithm's positions.")
-
-    if len(missing_stocks) > 0:
-
-        if len(missing_stocks) > 5:
+    (returns,
+     positions,
+     factor_returns,
+     factor_loadings) = _truncate_and_warn(returns,
+                                           positions,
+                                           factor_returns,
+                                           factor_loadings,
+                                           transactions=transactions,
+                                           pos_in_dollars=pos_in_dollars)
 
-            missing_stocks_displayed = (
-                " {} assets were missing factor loadings, including: {}..{}"
-            ).format(len(missing_stocks),
-                     ', '.join(missing_stocks[:5].map(str)),
-                     missing_stocks[-1])
-            avg_allocation_msg = "selected missing assets"
+    # Note that we convert positions to percentages *after* the checks
+    # above, since get_turnover() expects positions in dollars.
+    positions = _stack_positions(positions, pos_in_dollars=pos_in_dollars)
 
-        else:
-            missing_stocks_displayed = (
-                "The following assets were missing factor loadings: {}."
-            ).format(list(missing_stocks))
-            avg_allocation_msg = "missing assets"
+    risk_exposures_portfolio = compute_exposures(positions, factor_loadings,
+                                                 stack_positions=False)
```

Comment:
Can you put each of these on their own line?

Reply:
put each on their own line